papermario/src/code_fa4c0_len_3bf0.c

```diff
@@ -2,35 +2,130 @@
 
 INCLUDE_ASM(code_fa4c0_len_3bf0, func_802D5B10);
 
-INCLUDE_ASM(code_fa4c0_len_3bf0, FadeOutMusic);
+s32 FadeOutMusic(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+    s32 itemID = get_variable(script, *ptrReadPos++);
+    s32* ptrNextPos = ptrReadPos++;
 
-INCLUDE_ASM(code_fa4c0_len_3bf0, SetMusicTrack);
+    return (set_music_track(itemID, -1, 0, get_variable(script, *ptrNextPos++), 8) != 0) * 2;
+}
+
+s32 SetMusicTrack(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+    s32 musicPlayer = get_variable(script, *ptrReadPos++);
+    s32 songID = get_variable(script, *ptrReadPos++);
+    s32 variation = get_variable(script, *ptrReadPos++);
+    s32 volume = get_variable(script, *ptrReadPos++);
+
+    return (set_music_track(musicPlayer, songID, variation, 0x1F4, (volume << 0x10) >> 0x10) != 0) * 2;
+}
 
 INCLUDE_ASM(code_fa4c0_len_3bf0, FadeInMusic);
 
-INCLUDE_ASM(code_fa4c0_len_3bf0, PlayAmbientSounds);
+s32 PlayAmbientSounds(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+    s32 soundID = get_variable(script, *ptrReadPos++);
+
+    return (play_ambient_sounds(soundID, 0xFA) != 0) * 2;
+}
+
+s32 PlaySound(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+    s32 soundID = get_variable(script, *ptrReadPos++);
+
+    play_sound(soundID);
+    return 2;
+}
+
+s32 func_802D617C(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+    s32 soundID = get_variable(script, *ptrReadPos++);
+    s32 value2 = get_variable(script, *ptrReadPos++);
+
+    _play_sound(soundID, value2 & 0xFF, 0, 0);
+    return 2;
+}
+
+s32 PlaySoundAt(script_context* script, s32 initialCall) {
+    s32* ptrReadPos = script->ptrReadPos;
+    s32 soundID = get_variable(script, *ptrReadPos++);
+    s32 value2 = get_variable(script, *ptrReadPos++);
+    s32 x = get_variable(script, *ptrReadPos++);
+    s32 y = get_variable(script, *ptrReadPos++);
+    s32 z = get_variable(script, *ptrReadPos++);
+
+    play_sound_at_position(soundID, value2, (f32) x, (f32) y, (f32) z);
```

Comment:
These manual casts here shouldn't be necessary since they're implicly done by the type of these function parameters.